docs/v1/api.md

```diff
@@ -0,0 +1,333 @@
+# Service Catalog API
+
+This document details a standard API, including a schema for standard
+Kubernetes-native resources, to:
+
+- Formalize an API around the service broker concept as a first-class Kubernetes
+  (or Kubernetes-like) API for all to develop against.
+- Follow established patterns in the Kubernetes core API for globally-available
+  and namespaced resources
+- Facilitate the development of Kubernetes-like controllers in support of the
+  API, without undermining competition or choice
+- Ensure a smooth, consistent UX for users of the API and implementors of the
+  controllers
+
+# API Precedents in Kubernetes
+
+Broadly speaking, the system detailed herein can list services and allow
+consumers to provision, bind, unbind and deprovision them.
+
+We've drawn inspiration from the
+[Persistent Volumes](http://kubernetes.io/docs/user-guide/persistent-volumes/)
+API and the [Ingress Controllers](http://kubernetes.io/docs/user-guide/ingress/)
+system. The remainder of this section explains these inspirations in further
+detail.
+
+Additionally, we've adhered, wherever possible, to Kubernetes's declarative
+API. That is, the resources we've defined herein describe the _desired end
+state_ of the applicable systems. Kubernetes and the service controller
+implementations (see below) are responsible for fulfilling that end state.
+
+## Persistent Volumes
+
+First, this problem space is similar to that of persistent volumes. We model
+the data structures herein after those in persistent volumes primarily because
+we believe that consistency is a virtue. More importantly, the community
+benefits when similar problems have similar solutions.
+
+Specifically, we believe that the following existing types provide a good model
+for the mechanics of how the services offered by a broker can be used in
+Kubernetes:
+
+- `StorageClass`
+- `PersistentVolume`
+- `PersistentVolumeClaim`
+
+## Ingress Controllers
+
+Today, operators choose their own implementation of an ingress controller.
+Kubernetes gives the operator the ability to specify any ingress controller,
+and the complete freedom to implement it themselves.
+
+Similarly, permitting operators to implement or select their own service catalog
+controller would encourage freedom of choice and diversity. The analog to an
+ingress controller for the service catalog system is called a _service catalog
+controller_. Like ingress, these controllers run inside a standard pod. Anyone
+can write and deploy a service catalog controller, with no changes necessary to
+the Kubernetes install.
+
+Service catalog controllers vary, however, from ingress controllers, because a
+cluster can consist of many different controllers, each of which handles actions
+on a subset of the available service-related resources. For example, a cluster
+can run a service controller to handle AWS RDS databases, and a completely
+separate one to handle AWS Elastic Load Balancer instances.
```

Comment:
While this may be accurate, ELBs feel like an awkward example. ELBs are something that may be provisioned by an ingress controller or AWS cloud provider. So, both because there's little or no reason to provision ELBs through a broker _and_ because this paragraph is actively making a comparison to ingress controllers, using ELBs as an example could be confusing or at least distracting. Can we just swap in almost any other thing here? Mongo? Redis?